### PR TITLE
Customise xrdp

### LINKED ISSUE #2086: Customise xrdp login screen
## :white_check_mark: Checklist

- [x] I have searched open and closed issues for duplicates.
- [x] This is a request for a new feature in the Data Safe Haven or an upgrade to an existing feature.
- [x] The feature is still missing in the [latest version](https://github.com/alan-turing-institute/data-safe-haven/releases).
- [x] I have read through the [documentation](https://alan-turing-institute.github.io/data-safe-haven/).
- [x] This isn't an open-ended question (open a [discussion](https://github.com/alan-turing-institute/data-safe-haven/discussions) if it is).

## :strawberry: Suggested change

We should add a custom login screen for xrdp as was done for v4 releases

## :steam_locomotive: How could this be done?

Edit xrdp.conf

### PR DESCRIPTION
**Pull request by jemrobinson**
## :white_check_mark: Checklist

- [x] You have given your pull request a meaningful title (_e.g._ `Enable foobar integration` rather than `515 foobar`).
- [x] You are targeting the appropriate branch. If you're not certain which one this is, it should be **`develop`**.
- [x] Your branch is up-to-date with the **target branch** (it probably was when you started, but it may have changed since then).

### :vertical_traffic_light: Depends on

n/a

### :arrow_heading_up: Summary

- Customise xrdp login screen
- Apply some xrdp-sesman settings to block root login and kill disconnected sessions

Screenshot 2024-08-06 at 08 26 18

### :closed_umbrella: Related issues

Closes #2086

### :microscope: Tests

Tested on fresh SRE deployment

### CODE REVIEW

File: data_safe_haven/infrastructure/programs/sre/data.py [hunk 1]
```diff
@@ -547,7 +547,7 @@ def __init__(
                 str(file_path.relative_to(desired_state_directory)),
             )
             for file_path in sorted(desired_state_directory.rglob("*"))
-            if file_path.is_file()
+            if file_path.is_file() and not file_path.name.startswith(".")
```
Comment (JimMadge) [inline data_safe_haven/infrastructure/programs/sre/data.py:550]: I don't think we want this. There are skeleton files which start with .

File: data_safe_haven/resources/workspace/ansible/desired_state.yaml [hunk 2]
```diff
@@ -100,6 +100,36 @@
...
+    - name: Copy xrdp settings
+      ansible.builtin.copy:
+        src: etc/xrdp/xrdp.ini
+        dest: /etc/xrdp/xrdp.ini
+        mode: '0644'
+
+    - name: Copy xrdp logo
+      ansible.builtin.copy:
+        src: usr/share/xrdp/dsh_logo_240x140_256color.bmp
+        dest: /usr/share/xrdp/dsh_logo_240x140_256color.bmp
```
Comment (JimMadge) [inline data_safe_haven/resources/workspace/ansible/desired_state.yaml]: We should write this to another location, maybe `/var` or `/usr/local/share`. `/usr/share` is managed by `apt`.

File: data_safe_haven/resources/workspace/ansible/files/usr/share/xrdp/dsh_logo_240x140_256color.bmp
Comment (JimMadge) [inline data_safe_haven/resources/workspace/ansible/files/usr/share/xrdp/dsh_logo_240x140_256color.bmp:1]: Is this the size limit? The image does look quite soft at this resolution.

**Review by JimMadge — CHANGES_REQUESTED** [review]

File: data_safe_haven/infrastructure/programs/sre/data.py (hunk 1 above)
Comment (JimMadge) [inline data_safe_haven/infrastructure/programs/sre/data.py:550]:
```suggestion
            if file_path.is_file()
```

Comment (JimMadge) [reply data_safe_haven/infrastructure/programs/sre/data.py:550]: I was wrong 😄

**Review by JimMadge — APPROVED** [review]

File: data_safe_haven/resources/workspace/ansible/files/usr/share/xrdp/dsh_logo_240x140_256color.bmp
Comment (jemrobinson) [reply data_safe_haven/resources/workspace/ansible/files/usr/share/xrdp/dsh_logo_240x140_256color.bmp:1]: Yes, I tried with a bigger file and it overlaps the edges of the box.

**Pull request merged**